xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.cpp

```diff
@@ -132,7 +132,6 @@ bool CLinuxRendererGLES::Configure(const VideoPicture &picture, float fps, unsig
   // setup the background colour
   m_clearColour = CServiceBroker::GetWinSystem()->UseLimitedColor() ? (16.0f / 0xff) : 0.0f;
 
-  if (picture.hasDisplayMetadata && picture.hasLightMetadata)
```

Comment:
I don't think this "fixes" the problem. Allowing incomplete metadata may result in a worse image as the MaxFALL and MaxCLL provide info about the HDR content.

This means that if the encoding contains a valid HDR10 (pq) or HLG flag in the `color_transfer` then it can switch HDR mode on without any other metadata.

Simply seeing the HDR logo light up on the tv doesn't mean the picture is correct.

Ref:

```
Static metadata: SMPTE ST 2086 (mastering display color volume), MaxFALL (maximum frame-average light level), and MaxCLL (maximum content light level)
```
https://en.wikipedia.org/wiki/HDR10

And
```
MaxFALL/MaxCLL is static metadata required for HDR10 content. 
```

https://professionalsupport.dolby.com/s/article/Calculation-of-MaxFALL-and-MaxCLL-metadata?language=en_US

Reply:
It can't look worse than any of the tone mapping methods. IMO we'd still want to switch the TV to HDR/BT.2020.

Reply:
Not to mention that DRM PRIME platforms, eg. RPI4 allow switching to HDR mode with those "incomplete metadata" videos. Why this should be different for Intel?

Reply:
Even the TV's internal player would switch the TV to HDR mode with those videos. But on Intel PC they are tone mapped to SDR and look horrible.

Reply:
[This video](https://www.dropbox.com/scl/fi/t086p6sn4dajhv6bpdhgt/Life-Untouched-4K-Demo.mp4?rlkey=dk5gdhhee30fr9sntlgtczq1d&dl=1), for example, doesn't even trigger the tone mapping and it is absolutely unwatchable. With this PR it would switch the TV to HDR mode and looks perfectly fine, same as with TV's internal player.

Reply:
> Not to mention that DRM PRIME platforms, eg. RPI4 allow switching to HDR mode with those "incomplete metadata" videos. Why this should be different for Intel?

If the behaviour is inconsistent across platforms we should check how other platforms and applications handle it.

If it's expected to be able to just ignore any metadata then so be it but we should investigate more about what the correct behaviour should be.

Reply:
I also removed metadata checks for tonemapping. HLG streams and HDR10 videos with missing MaxFALL/MaxCLL didn't trigger tonemapping for SDR TVs.

Reply:
I was going to suggest removing the check altogether if you hadn't done it already. For Windows, HDR mode is triggered by color_transfer + primaries only. Metadata, when present, is forwarded as is to the TV, it hasn't caused issues that I know of. But it's possible that Windows is interfering in the process.

HLG streams typically don't come with metadata and that's valid. Windows has special handling though because of missing HLG passthrough support, Instead it's converted to PQ with shader and metadata is made up with 1000nits max, value found in many places of HLG spec.